### PR TITLE
Noodling

### PR DESCRIPTION
**Pull request by jdinan**
* Update Travis to run "hello" unit test for transport-none builds
* Update sync to be communication-free for single PE runs to support ^^^
* Add threading level and number of PEs to debugging output
* Fix Portals 4 and on-node comms bug

### CODE REVIEW

**Comment by jdinan** [comment]
Portals build (--with-cma --enable-error-checking --enable-profiling --disable-mr-scalable --enable-av-map --enable-remote-virtual-addressing) failing to run `hello`:
```
$ mpiexec -np 1 test/unit/hello
Sandia OpenSHMEM 1.4.0
  SHMEM_INFO                 1 (type: bool, default: 0)
	Print library information message at startup
  SHMEM_VERSION              0 (type: bool, default: 0)
	Print library version at startup
  SHMEM_DEBUG                0 (type: bool, default: 0)
	Enable debugging messages
  SHMEM_SYMMETRIC_SIZE       536870912 (type: size, default: 536870912)
	Symmetric heap size

Additional options:
  SHMEM_SYMMETRIC_HEAP_USE_HUGE_PAGES 0 (type: bool, default: 0)
	Use Linux huge pages for symmetric heap
  SHMEM_SYMMETRIC_HEAP_PAGE_SIZE 2097152 (type: size, default: 2097152)
	Page size to use for huge pages
  SHMEM_SYMMETRIC_HEAP_USE_MALLOC 0 (type: bool, default: 0)
	Allocate the symmetric heap using malloc
  SHMEM_BOUNCE_SIZE          2048 (type: size, default: 2048)
	Maximum message size to bounce buffer
  SHMEM_MAX_BOUNCE_BUFFERS   128 (type: long, default: 128)
	Maximum number of bounce buffers per context
  SHMEM_TRAP_ON_ABORT        0 (type: bool, default: 0)
	Generate trap if the program aborts or calls shmem_global_exit

Collectives options:
  SHMEM_COLL_CROSSOVER       4 (type: long, default: 4)
	Crossover between linear and tree collectives
  SHMEM_COLL_RADIX           4 (type: long, default: 4)
	Radix for tree-based collectives
  SHMEM_BARRIER_ALGORITHM    auto (type: string, default: auto)
	Algorithm for barrier.  Options are auto, linear, tree, dissem
  SHMEM_BCAST_ALGORITHM      auto (type: string, default: auto)
	Algorithm for broadcast.  Options are auto, linear, tree
  SHMEM_REDUCE_ALGORITHM     auto (type: string, default: auto)
	Algorithm for reductions.  Options are auto, linear, tree, recdbl
  SHMEM_COLLECT_ALGORITHM    auto (type: string, default: auto)
	Algorithm for collect.  Options are auto, linear
  SHMEM_FCOLLECT_ALGORITHM   auto (type: string, default: auto)
	Algorithm for fcollect.  Options are auto, linear, ring, recdbl

Network transport: Portals 4

On-node transport: Linux CMA
  SHMEM_CMA_PUT_MAX          8192 (type: size, default: 8192)
	Size below which to use CMA for puts
  SHMEM_CMA_GET_MAX          16384 (type: size, default: 16384)
	Size below which to use CMA for gets

[0000] ERROR: ../../src/transport_portals4.c:677: shmem_transport_startup
[0000]        PtlLEAppend of all memory failed: 1
[0000] ERROR: ../../src/init.c:298: shmem_internal_init
[0000]        Transport startup failed (1)

===================================================================================
=   BAD TERMINATION OF ONE OF YOUR APPLICATION PROCESSES
=   PID 20843 RUNNING AT travis-job-215e9d2b-37d0-4700-8a57-3e996cc52963
=   EXIT CODE: 134
=   CLEANING UP REMAINING PROCESSES
=   YOU CAN IGNORE THE BELOW CLEANUP MESSAGES
===================================================================================
YOUR APPLICATION TERMINATED WITH THE EXIT STRING: Aborted (signal 6)
This typically refers to a problem with your application.
Please see the FAQ page for debugging suggestions
```

**Comment by jdinan** [comment]
Appears to be related to combining CMA and Portals.

**Comment by jdinan** [comment]
Oh boy, CMA and XPMEM haven't worked with Portals 4 since v1.2.0.  Should be fixed now..

**Comment by jdinan** [comment]
Here's a sample of the updated DEBUG output (the line with thread level is new):
```
[0000] DEBUG: ../../src/init.c:254: shmem_internal_init
[0000]        Thread level=SINGLE, Num. PEs=1
[0000]        Sym. heap=0x80000000 len=537919488 -- data=0x600da8 len=24
```

**Comment by jdinan** [comment]
@wrrobin Could you please review this PR?

**Comment by wrrobin** [comment]
Sure @jdinan. Would take a look at this soon.

File: .travis.yml [hunk 1]
```diff
@@ -10,7 +10,8 @@ env:
     global:
         - TRAVIS_PAR_MAKE="-j 4"
         - SOS_GLOBAL_BUILD_OPTS="--enable-picky --enable-pmi-simple FCFLAGS=-fcray-pointer"
-        - PTL_IFACE_NAME=venet0
+        #- PTL_IFACE_NAME=venet0
```
Comment (wrrobin) [inline .travis.yml]: Do we need to keep this commented line?

Comment (jdinan) [reply .travis.yml]: I left it there as a note, since I don't think I'll remember it if we ever decide to enable more Portals testing in Travis.

Comment (jdinan) [reply .travis.yml]: For some reason, Portals isn't working on venet0 any more so I set it to use loopback, which is fine for single PE tests.

Comment (wrrobin) [reply .travis.yml]: ok, thanks

**Pull request merged**